HeartbeatLogging/Tests/Unit/StorageTests.swift

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import HeartbeatLogging
+
+private let data = "test_data".data(using: .utf8)!
+
+class FileStorageTests: XCTestCase {
+  func testRead_WhenFileDoesNotExist_ThrowsError() throws {
+    // Given
+    let fileStorage = FileStorage(url: makeTemporaryFileURL())
+    // Then
+    XCTAssertThrowsError(try fileStorage.read())
+  }
+
+  func testRead_WhenFileExists_ReturnsFileContents() throws {
+    // Given
+    let fileStorage = FileStorage(url: makeTemporaryFileURL())
+
+    XCTAssertNoThrow(try fileStorage.write(data))
+    // When
+    let storedData = try fileStorage.read()
+    // Then
+    XCTAssertEqual(storedData, data)
+  }
+
+  func testWriteData_WhenFileDoesNotExist_CreatesFile() throws {
+    // Given
+    let fileStorage = FileStorage(url: makeTemporaryFileURL())
+    XCTAssertThrowsError(try fileStorage.read())
+    // When
+    XCTAssertNoThrow(try fileStorage.write(data))
+    // Then
+    let storedData = try fileStorage.read()
+    XCTAssertEqual(storedData, data)
+  }
+
+  func testWriteData_WhenFileExists_ModifiesFile() throws {
+    // Given
+    let fileStorage = FileStorage(url: makeTemporaryFileURL())
+
+    XCTAssertNoThrow(try fileStorage.write(data))
+    // When
+    let modifiedData = "modified_data".data(using: .utf8)
+    XCTAssertNoThrow(try fileStorage.write(modifiedData))
+
+    // Then
+    let storedData = try fileStorage.read()
+    XCTAssertEqual(storedData, modifiedData)
+  }
+
+  func testWriteNil_WhenFileDoesNotExist_DoesNothing() throws {
+    // Given
+    let fileStorage = FileStorage(url: makeTemporaryFileURL())
+    XCTAssertThrowsError(try fileStorage.read())
+    // When
+    XCTAssertNoThrow(try fileStorage.write(nil))
```

Comment:
Why does write(nil) remove versus write an empty file?

I would expect `remove` to be a different API and `write` nil to be an empty operation.

Reply:
Ok, thanks for the input. I'll rework this to make `write(nil)` empty the file rather than remove it.

Reply:
I reworked things in 26f5246 so that write(nil) is an operation that write empty data to the storage.